File: pset6/mario.py

import cs50
import sys

def mario():

    print("Enter pyramid height(between 0 and 23):", end=' ')
    x=cs50.get_int()
        
    bool1=False
    if x<0 or x>23:
        bool1=True
    
    while bool1==True:
        print("Enter pyramid height(between 0 and 23):", end=' ')
        x=cs50.get_int()
        if x>=0 and x<=23:
            break
    
    
    i=0
    j=0
    y=0
    
    if x>=0 and x<=23:

        for i in range(x):
        
            for j in range(x-i-1):
                print(" ", end='')
            for y in range(i+2):
                print("#", end='')
            print("")
            
        
        
if __name__ == "__main__":
    mario()

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>
int main(void)
{   
    int i =0;
    
    
    printf("Enter pyramid height(between 0 and 23):\n");
    int x = GetInt();
        while ((x > 23) || (x < 0)) {
         printf("Enter pyramid height(between 0 and 23):\n");
            x = GetInt();
        }
        
    
    
    if (x>=0 && x<=23) {
    while (i<=x) {
        
    for (i=1;i<=x;i++) {
        for (int j=1;j<=(x-i);j++) {
            printf(" ");
        }
            for (int y=1; y<= i+1; y++) {
                printf("#");
            }
            printf("\n");
    }
    } 
} else { printf("Enter pyramid height(between 0 and 23):\n");
            x = GetInt();
    do {
        
    for (i=1;i<=x;i++) {
        for (int j=1;j<=(x-i);j++) {
            printf(" ");
        }
            for (int y=1; y<= i+1; y++) {
                printf("#");
            }
            printf("\n");
    }
    } while (i<=x);
}
}